servers/mcp-neo4j-memory/src/mcp_neo4j_memory/server.py

```diff
@@ -14,456 +14,556 @@
 import mcp.server.stdio
 
 # Set up logging
-logger = logging.getLogger('mcp_neo4j_memory')
+logger = logging.getLogger("mcp_neo4j_memory")
 logger.setLevel(logging.INFO)
 
+
 # Models for our knowledge graph
 class Entity(BaseModel):
-    name: str
-    type: str
-    observations: List[str]
+  name: str
+  type: str
+  observations: List[str]
+
 
 class Relation(BaseModel):
-    source: str
-    target: str
-    relationType: str
+  source: str
+  target: str
+  relationType: str
+
 
 class KnowledgeGraph(BaseModel):
-    entities: List[Entity]
-    relations: List[Relation]
+  entities: List[Entity]
+  relations: List[Relation]
+
 
 class ObservationAddition(BaseModel):
-    entityName: str
-    contents: List[str]
+  entityName: str
+  contents: List[str]
+
 
 class ObservationDeletion(BaseModel):
-    entityName: str
-    observations: List[str]
+  entityName: str
+  observations: List[str]
+
 
 class Neo4jMemory:
-    def __init__(self, neo4j_driver):
-        self.neo4j_driver = neo4j_driver
-        self.create_fulltext_index()
-
-    def create_fulltext_index(self):
-        try:
-            # TODO , 
-            query = """
-            CREATE FULLTEXT INDEX search IF NOT EXISTS FOR (m:Memory) ON EACH [m.name, m.type, m.observations];
-            """
-            self.neo4j_driver.execute_query(query)
-            logger.info("Created fulltext search index")
-        except neo4j.exceptions.ClientError as e:
-            if "An index with this name already exists" in str(e):
-                logger.info("Fulltext search index already exists")
-            else:
-                raise e
-
-    async def load_graph(self, filter_query="*"):
-        query = """
+  def __init__(self, neo4j_driver):
+    self.neo4j_driver = neo4j_driver
+    self.create_fulltext_index()
+
+  def create_fulltext_index(self):
+    """Create fulltext indexes for nodes and relationships in the Neo4j graph."""
+    index_config = {
```

Comment:
I think this is overkill for default usage
changing the default tokenizer is not people would expect
also this is not a terabyte graph :) that needs eventual consistency